expression/builtin_compare_vec.go

```diff
@@ -471,11 +471,50 @@ func (b *builtinIntervalRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTRealSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err = b.args[0].VecEvalReal(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err = b.args[1].VecEvalReal(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	i64s := result.Int64s()
+	x := buf0.Float64s()
+	y := buf1.Float64s()
+	for i := 0; i < n; i++ {
+		isNull0 := buf0.IsNull(i)
+		isNull1 := buf1.IsNull(i)
+		if isNull0 || isNull1 {
+			if compareNull(isNull0, isNull1) < 0 {
```

Comment:
The row-based implementation returns null here and ignores`compareNull`.

https://github.com/pingcap/tidb/blob/a22af3f821b2798285f2ae98b23f22de7dfde8b5/expression/builtin_compare.go#L2436

If I have not missed something, please take a further look that why the test is passed.

Reply:
The test does not check the value in the null case. So it need to skip this case?

https://github.com/fzhedu/tidb/blob/0fbf7fc57f5049a96735b8099734e0118550c355/expression/bench_test.go#L773